src/app/link.service.ts

```diff
@@ -26,38 +28,73 @@ export interface LinkData {
 })
 export class LinkService {
 
+  private readonly noConnectionErrorMessage = `Parent site was not found,\
+  please try reloading the parent site and relaunching the extension`;
+
   private parentSource = new BehaviorSubject<string>('');
   parent$ = this.parentSource.asObservable();
 
   private linkListSource = new BehaviorSubject<LinkData[]>([]);
   linkList$ = this.linkListSource.asObservable();
 
-  private setLinks(newLinks: LinkData[]): void {
-    this.linkListSource.next(newLinks);
-  }
+  private parentTabId: number;
 
   constructor(private ngZone: NgZone, private title: Title) {
 
-    chrome.tabs.getCurrent((tab) => {
-      chrome.tabs.sendMessage(tab.openerTabId, {
-        message: 'send parent'
-      }, (parent: string) => {
-        this.ngZone.run(() => {
-          this.parentSource.next(parent);
-        });
+    chrome.tabs.getCurrent((currTab: chrome.tabs.Tab) => {
+      this.parentTabId = currTab.openerTabId;
+      chrome.windows.create({
+        tabId: currTab.id,
+        left: 50,
+        top: 50
+      }, (window: chrome.windows.Window) => {
+        this.requestLinkData();
       });
+    });
+  }
 
-      chrome.tabs.sendMessage(tab.openerTabId, {
-        message: 'send links'
-      }, (links: LinkData[]) => {
+  private requestLinkData() {
+
+    chrome.tabs.sendMessage(this.parentTabId, {
+      message: 'send link data'
+    }, (response) => {
+      if (response === undefined) {
+        alert(this.noConnectionErrorMessage);
+        console.error(this.noConnectionErrorMessage);
+      } else if (!response.success) {
+        console.error(response.errorMessage);
+      } else {
         this.ngZone.run(() => {
-          this.setLinks(links);
+          this.setLinks(response.linkList);
         });
-      });
+      }
+    });
 
-      chrome.tabs.get(tab.openerTabId, (parentTab) => {
-        this.title.setTitle(parentTab.title + ' (Link Diver)');
+    chrome.tabs.get(this.parentTabId, (parentTab) => {
+      this.title.setTitle(parentTab.title + ' (Link Diver)');
+      this.parentSource.next(parentTab.url);
+    });
+  }
+
+  private setLinks(newLinks: LinkData[]): void {
+    this.linkListSource.next(newLinks);
+  }
+
+  highlightLink(link: LinkData): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      chrome.tabs.sendMessage(this.parentTabId, {
+        message: 'highlight link',
+        linkData: link
+      }, (response) => {
+        if (response === undefined) {
```

Comment:
Nit: I think we can do this too

if (!response) {
}

It will handle null response and undefined response.